coriolis/tests/osmorphing/test_windows.py

```diff
@@ -0,0 +1,893 @@
+            "'%(service_account)s' -Type String -Force;"
+            "New-ItemProperty -Path '%(path)s' -Name 'Start' -Value "
+            "%(start_mode)s -Type DWord -Force;"
+            "New-ItemProperty -Path '%(path)s' -Name 'Type' -Value "
+            "16 -Type DWord -Force;"
+            "New-ItemProperty -Path '%(path)s' -Name 'ErrorControl' -Value "
+            "0 -Type DWord -Force" %
+            {"path": registry_path, "image_path": mock.sentinel.image_path,
+             "display_name": mock.sentinel.display_name,
+             "description": mock.sentinel.description,
+             "depends_on": depends_on_ps, "service_account": service_account,
+             "start_mode": windows.SERVICE_START_AUTO}
+        )
+
+        self.conn.exec_ps_command.assert_called_once_with(
+            expected_commands, ignore_stdout=True)
+
+    @mock.patch.object(windows.utils, 'write_winrm_file')
+    def test_run_user_script(self, mock_write_winrm_file):
+        user_script = 'echo "Hello, World!"'
+        script_path = "$env:TMP\\coriolis_user_script.ps1"
+
+        self.morphing_tools.run_user_script(user_script)
+
+        mock_write_winrm_file.assert_called_once_with(
+            self.conn, script_path, user_script)
+        self.conn.exec_ps_command.assert_called_once_with(
+            ('$ErrorActionPreference = "Stop"; powershell.exe '
+             '-NonInteractive -ExecutionPolicy RemoteSigned '
+             '-File "%(script)s" "%(os_root_dir)s"') % {
+                 "script": script_path,
+                 "os_root_dir": self.os_root_dir,
+            }
+        )
+
+    @mock.patch.object(windows.utils, 'write_winrm_file')
+    def test_run_user_script_empty_script(self, mock_write_winrm_file):
+        result = self.morphing_tools.run_user_script("")
+
+        self.assertIsNone(result)
+        mock_write_winrm_file.assert_not_called()
+        self.conn.exec_ps_command.assert_not_called()
+
+    @mock.patch.object(windows.utils, 'write_winrm_file')
+    def test_run_user_script_raises_exception_on_write_winrm_file(
+            self, mock_write_winrm_file):
+        user_script = 'echo "Hello, World!"'
+        mock_write_winrm_file.side_effect = exception.CoriolisException
+
+        self.assertRaises(
+            exception.CoriolisException,
+            self.morphing_tools.run_user_script, user_script)
+
+        self.conn.exec_ps_command.assert_not_called()
+
+    @mock.patch.object(windows.utils, 'write_winrm_file')
+    def test_run_user_script_raises_exception_on_exec_ps_command(
+            self, mock_write_winrm_file):
+        user_script = 'echo "Hello, World!"'
+
+        self.conn.exec_ps_command.side_effect = exception.CoriolisException
+
+        self.assertRaises(
+            exception.CoriolisException,
+            self.morphing_tools.run_user_script, user_script)
+
+        mock_write_winrm_file.assert_called_once_with(
+            self.conn, mock.ANY, user_script)
```

Comment:
Make the assertion stronger by using this:
```python
script_path = "$env:TMP\\coriolis_user_script.ps1"
...
...
...
mock_write_winrm_file.assert_called_once_with(
    self.conn, script_path, user_script)
```
instead of `mock.ANY`.